beacon-chain/core/capella/transition.go

```diff
@@ -0,0 +1,54 @@
+package capella
+
+import (
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	enginev1 "github.com/prysmaticlabs/prysm/v3/proto/engine/v1"
+)
+
+// ETH1AddressOffset is the offset at which the execution address starts
+// within an ETH1-prefixed withdrawal credential
+const ETH1AddressOffset = 12
+
+// WithdrawBalance withdraws the balance from the validator and creates a
+// withdrawal receipt for the EL to process
+func withdrawBalance(pre state.BeaconState, index types.ValidatorIndex, amount uint64) (state.BeaconState, error) {
+	b := pre.Copy()
```

Comment:
Having a copy a state per every validator withdrawal is excessive! I think we need a better solution here
It's ok to move forward but I recommend that we add a TODO here or add an issue to track this